library/src/main/java/org/pollub/library/item/controller/MovieController.java

```diff
@@ -0,0 +1,76 @@
+package org.pollub.library.item.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.pollub.library.item.model.MovieDisc;
+import org.pollub.library.item.model.dto.MovieCreateDto;
+import org.pollub.library.item.service.IMovieService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/movie")
+@RequiredArgsConstructor
+public class MovieController {
+    private final IMovieService movieService;
+
+    @PostMapping
+    public ResponseEntity<MovieDisc> createMovie(@Valid @RequestBody MovieCreateDto dto) {
+        return ResponseEntity.ok(movieService.createMovie(dto));
+    }
+    @GetMapping("/{id}")
+    public ResponseEntity<MovieDisc> getMovie(@PathVariable Long id) {
+        return ResponseEntity.ok(movieService.findById(id));
+    }
+
+    @GetMapping("/director/{director}")
+    public ResponseEntity<List<MovieDisc>> getMoviesByDirector(@PathVariable String director) {
+        return ResponseEntity.ok(movieService.findByDirector(director));
+    }
+
+    @GetMapping("/genre/{genre}")
+    public ResponseEntity<List<MovieDisc>> getMoviesByGenre(@PathVariable String genre) {
+        return ResponseEntity.ok(movieService.findByGenre(genre));
+    }
+
+    @GetMapping("/fileFormat/{fileFormat}")
+    public ResponseEntity<List<MovieDisc>> getMoviesByFileFormat(@PathVariable String fileFormat) {
+        return ResponseEntity.ok(movieService.findByFileFormat(fileFormat));
+    }
+
+    @GetMapping("/resolution/{resolution}")
+    public ResponseEntity<List<MovieDisc>> getMoviesByResolution(@PathVariable String resolution) {
+        return ResponseEntity.ok(movieService.findByResolution(resolution));
+    }
+
+    @GetMapping("/duration/{minDuration}/{maxDuration}")
```

Comment:
If you use @RequestParam then its gonna be like this
http://localhost:8080/api/movie?title=abc&director=asdad
not like you did /abc/asdds thats called @PathVariable
so remove /{title}/{director}

Reply:
fixed